### PR TITLE
PS-3706 base type only

### PR DESCRIPTION
**Pull request by pivnicek**
refactored from #16  to just use base types

### CODE REVIEW

**Comment by pivnicek** [comment]
cau @ErikZigo, this is the new PR replacing #16 as the first step.

**Comment by pivnicek** [comment]
ah, zapomel jsem case kdyz to nema definovane `column_metadata` ale ma `columns`, jdu na to

**Comment by ErikZigo** [comment]
@pivnicek please assign me as reviewer when it will be ready for that. Thanks.

**Comment by pivnicek** [comment]
@ErikZigo I caught an issue there regarding PKs, but if you could take a look and give me some advice on the other parts (I'm most concerned about missing test cases etc...) that would help me a lot.  Thanks, sorry for being so late.

**Comment by ErikZigo** [comment]
I will look on it. At the first I see that not all of code is covered by unit test. Please try to fix it.

**Review by ErikZigo — COMMENTED** [review]
I will look on it. At the first I see that not all of code is covered by unit test. Please try to fix it.

**Comment by pivnicek** [comment]
Yeah, sorry about that @ErikZigo , all classes are covered now.

File: libs/output-mapping/src/Writer/TableWriter.php [hunk 1]
```diff
@@ -67,11 +71,17 @@ public function setFormat($format)
...
      * @throws \Exception
      */
-    public function uploadTables($sourcePathPrefix, array $configuration, array $systemMetadata, $stagingStorageOutput)
-    {
+    public function uploadTables(
+        $sourcePathPrefix,
+        array $configuration,
+        array $systemMetadata,
+        $stagingStorageOutput,
+        $typedTableEnabled = false
```
Comment (odinuv) [inline libs/output-mapping/src/Writer/TableWriter.php:83]: To pojmenování mi nepřijde úplně dobrý co třeba "createTypedTables" ?

File: libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionFactory.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Writer\Table\TableDefinition;
+
+class TableDefinitionFactory
```
Comment (odinuv) [inline libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionFactory.php:7]: moc nechápu k čemu je tu tahle factory

File: libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumnFactory.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Writer\Table\TableDefinition;
+
+use Keboola\Datatype\Definition\Common;
+
+class TableDefinitionColumnFactory
```
Comment (odinuv) [inline libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumnFactory.php:9]: a vlastně ani tahle :) ale jestli to @ErikZigo přijde v pohodě, tak 🤷

**Review by odinuv — APPROVED** [review]

File: libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionFactory.php (hunk 2 above)
Comment (pivnicek) [reply libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionFactory.php:7]: uvidis az budem supportovat ten column `Definition` param ;-)

File: libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumnFactory.php (hunk 3 above)
Comment (pivnicek) [reply libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumnFactory.php:9]: jj, to stejny.  muzes vidit proc je to takhle na ten posledni PR

Comment (ErikZigo) [reply libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumnFactory.php:9]: Nechme tak

**Pull request merged**